### PR TITLE
Optimize logic in EvenPodsSpread API validation

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/kind cleanup
/sig scheduling

**What this PR does / why we need it**:

A followup of https://github.com/kubernetes/kubernetes/pull/77327#discussion_r304984621.

cc/ @alculquicondor.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/lgtm

File: pkg/apis/core/validation/validation.go [hunk 1]
```diff
@@ -5627,12 +5615,12 @@ func ValidateWhenUnsatisfiable(fldPath *field.Path, action core.UnsatisfiableCon
 	return nil
 }
 
-// ValidateSpreadConstraintPair tests that if `pair` exists in `existingConstraintPairs`.
-func ValidateSpreadConstraintPair(fldPath *field.Path, pair spreadConstraintPair, existingConstraintPairs []spreadConstraintPair) *field.Error {
-	for _, existingPair := range existingConstraintPairs {
-		if pair.topologyKey == existingPair.topologyKey &&
-			pair.whenUnsatisfiable == existingPair.whenUnsatisfiable {
-			return field.Duplicate(fldPath, pair)
+// ValidateSpreadConstraintNotRepeat tests that if `pair` exists in `existingConstraintPairs`.
```
Comment (alculquicondor) [inline pkg/apis/core/validation/validation.go]: The comment is now out of date.

File: pkg/apis/core/validation/validation.go [hunk 2]
```diff
@@ -5627,12 +5615,12 @@ func ValidateWhenUnsatisfiable(fldPath *field.Path, action core.UnsatisfiableCon
...
+func ValidateSpreadConstraintNotRepeat(fldPath *field.Path, constraint core.TopologySpreadConstraint, restingConstraints []core.TopologySpreadConstraint) *field.Error {
```
Comment (alculquicondor) [inline pkg/apis/core/validation/validation.go]: Why pass fldPath instead of putting it statically in this function.

Comment (Huang-Wei) [reply pkg/apis/core/validation/validation.go]: I'd say that's a convention to encapsulate an error in a stateless manner.

File: pkg/apis/core/validation/validation.go (hunk 1 above)
Comment (Huang-Wei) [reply pkg/apis/core/validation/validation.go]: Updated.

**Comment by alculquicondor** [comment]
LGTM

**Comment by Huang-Wei** [comment]
/cc @lavalamp 

A small optimization to eliminate some temporary variables.

**Comment by Huang-Wei** [comment]
cc/ @lavalamp @liggitt Does the change make sense to you?

**Pull request merged**